### PR TITLE
Add stylelint to the circlecli

### PR DESCRIPTION
**Pull request by lukasholzer**
### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by lukasholzer** [comment]
If the stacked-bar-chart PR gets merged rebase this one and refactor the `stylelint` target.

#988

File: .circleci/config.yml [hunk 1]
```diff
@@ -440,9 +480,8 @@ workflows:
       - lint-workspace:
           requires:
             - install
-      - lint-styles:
-          requires:
-            - install
+      - bazel_stylelint:
+          <<: *filter_forks
```
Comment (lukasholzer) [inline .circleci/config.yml:484]: Why do we filter here for forked pull requests?

File: .circleci/config.yml [hunk 2]
```diff
@@ -139,21 +146,33 @@ commands:
...
+      - run:
+          name: Setting up Bazel configuration for CI
+          command: echo "import %workspace%/.circleci/bazel.rc" >> ./.bazelrc
       - run:
-          name: Setup Bazel CI
-          command: npm install -G @bazel/bazelisk && cd tools/bazel_build_cache && sh setup_ci_build_cache.sh
+          name: Setup Bazel Build Cache
+          command: cd ./tools/bazel_build_cache && sh setup_ci_build_cache.sh
+      # npm ci --ignore-scripts &> /dev/null
+      #- run: npm ci --ignore-scripts
```
Comment (lukasholzer) [inline .circleci/config.yml:161]:
```suggestion
```

File: WORKSPACE [hunk 3]
```diff
@@ -81,6 +66,8 @@ npm_install(
     name = "npm",
     package_json = "//:package.json",
     package_lock_json = "//:package-lock.json",
+    symlink_node_modules = True,
```
Comment (lukasholzer) [inline WORKSPACE:69]: When we symlink the node_modules can we integrate them in the docker image?

File: libs/shared/design-system/ui/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,36 @@
+load("@npm_bazel_typescript//:index.bzl", "ts_config", "ts_library")
+
+package(default_visibility = ["//visibility:public"])
```
Comment (lukasholzer) [inline libs/shared/design-system/ui/BUILD.bazel:3]:
```suggestion
package(default_visibility = ["//:__subpackages__"])
```

File: WORKSPACE (hunk 3 above)
Comment (furtnerl) [reply WORKSPACE:69]: We can't integrate them in a clean way either way. I have checked both options and True is the recommended default value there

File: .circleci/config.yml (hunk 1 above)
Comment (furtnerl) [reply .circleci/config.yml:484]: good catch, I don't see a reason why this filter should be applied, will change

File: libs/shared/design-system/ui/BUILD.bazel (hunk 4 above)
Comment (furtnerl) [reply libs/shared/design-system/ui/BUILD.bazel]: done

File: .circleci/config.yml (hunk 1 above)
Comment (furtnerl) [reply .circleci/config.yml:480]: just rebased proposed changes again

File: .dockerignore [hunk 5]
```diff
@@ -3,7 +3,7 @@
 #--------------------------
 **
 
-#--------------------------
+#--------------------------a
```
Comment (ffriedl89) [inline .dockerignore:6]:
```suggestion
#--------------------------
```

Comment (ffriedl89) [reply .dockerignore:6]: :D

**Pull request merged**